### ISSUE TITLE
Doesn't work with federated schema

### ISSUE THREAD

**Issue by yuhanz** [kamilkisiela/graphql-inspector#1614]
Try this schema with graphql federation:
https://github.com/apollographql/federation-jvm/blob/master/spring-example/src/main/resources/schemas/inventory.graphql

`graphql-inspector diff /tmp/inventory.graphql /tmp/inventory.graphql`

```
This resulted in the following error:

error Error: Unknown directive "@key".

Unknown directive "@extends".

Unknown directive "@external".
    at assertValidSDL (/usr/local/lib/node_modules/graphql/validation/validate.js:107:11)
    at Object.buildASTSchema (/usr/local/lib/node_modules/graphql/utilities/buildASTSchema.js:45:34)
    at makeSchema (/usr/local/lib/node_modules/@graphql-inspector/cli/node_modules/@graphql-tools/merge/index.cjs.js:954:99)
    at Object.mergeSchemasAsync (/usr/local/lib/node_modules/@graphql-inspector/cli/node_modules/@graphql-tools/merge/index.cjs.js:945:12)
    at async Object.loadSchema (/usr/local/lib/node_modules/@graphql-inspector/cli/node_modules/@graphql-tools/load/index.cjs.js:650:20)
Error: Unknown directive "@key".

Unknown directive "@extends".

Unknown directive "@external".
    at assertValidSDL (/usr/local/lib/node_modules/graphql/validation/validate.js:107:11)
    at Object.buildASTSchema (/usr/local/lib/node_modules/graphql/utilities/buildASTSchema.js:45:34)
    at makeSchema (/usr/local/lib/node_modules/@graphql-inspector/cli/node_modules/@graphql-tools/merge/index.cjs.js:954:99)
    at Object.mergeSchemasAsync (/usr/local/lib/node_modules/@graphql-inspector/cli/node_modules/@graphql-tools/merge/index.cjs.js:945:12)
    at async Object.loadSchema (/usr/local/lib/node_modules/@graphql-inspector/cli/node_modules/@graphql-tools/load/index.cjs.js:650:20)
```

graphq-inspector version 2.1.0

**Comment by jinhong-** [comment]
I am also looking at this right now. Most likely a PR can be submitted, similar to PR #505? The specifications for federated schema is [here](https://www.apollographql.com/docs/apollo-server/federation/federation-spec/#federation-schema-specification). Looks straightfoward enough to add more directives by modifying the ```extendSchema``` function. @kamilkisiela are you accepting PRs for this?

**Comment by kamilkisiela** [comment]
YES :) I would do it anyway but you're probably going to make it sooner than me :)

**Comment by jinhong-** [comment]
Submitted #1661. Probably need some suggestions on how to improve the code. I realized that the validation errors come from loading the schema. I merged the federation schema in as the schema is getting loaded

**Comment by Jastrzebowski** [comment]
I saw the PR was merged, @kamilkisiela any idea when it will be released, docs mentioning [`--federation`](https://graphql-inspector.com/docs/essentials/serve) are already published.

**Comment by scripter-co** [comment]
@kamilkisiela do you have an update on when you're thinking about publishing the federation changes?

**Comment by kamilkisiela** [comment]
Ok, released `2.3.0`
